ntpcheck/checker/runner.go

```diff
@@ -104,6 +103,31 @@ func RunCheck(address string) (*NTPCheckResult, error) {
 	return checker.Run()
 }
 
+// RunNTPData is a simple wrapper to connect to address and run NTPCheck.Run()
+// If using chrony it gathers extra info about the peers using the unix socket
+func RunNTPData(address string) (*NTPCheckResult, error) {
+	timeout := 5 * time.Second
+	deadline := time.Now().Add(timeout)
+	flavour := getFlavour()
+	if address == "" {
```

Comment:
Let's swap those lines, so we don't decide on the address for `RunCheck`
```suggestion
	if flavour != flavourChrony {
		return RunCheck(address)
	}
	if address == "" {
		address = getPrivateServer(flavour)
	}
```